core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -783,6 +783,13 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val CACHE_CHECKPOINT_PREFERRED_LOCS =
+    ConfigBuilder("spark.rdd.checkpoint.cachePreferredLocs")
+      .internal()
+      .doc("Whether to cache preferred locations of checkpointed RDD.")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
What is the side-effect of this feature?

Reply:
I can not think of possible side-effect for now. To be conservative, we can have false as default.

Reply:
Then the side-effect should be, when the cached preferred locations are outdated, the tasks launched for the partitions might be at non optimal hosts.